From: Simon McVittie <smcv@debian.org>
Date: Mon, 30 Jul 2018 16:51:01 +0100
Subject: Skip test-pull-repeated during CI

This test is expected to fail a small proportion of the time. During
the build of ostree 2018.7-1 in Debian, it seems we were unlucky on
s390x. Non-deterministic tests are also problematic for autopkgtest,
where they can gate migration of our dependencies like GLib, so skip
this test unless the caller has opted-in to non-deterministic tests.

It would be appropriate to enable this test in environments where
failures can easily be retried and are not disruptive to other
packages.

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 tests/test-pull-repeated.sh | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tests/test-pull-repeated.sh b/tests/test-pull-repeated.sh
index 237a863..c646c13 100755
--- a/tests/test-pull-repeated.sh
+++ b/tests/test-pull-repeated.sh
@@ -23,6 +23,10 @@ set -euo pipefail
 
 . $(dirname $0)/libtest.sh
 
+if [ "${OSTREE_TEST_ALLOW_RANDOM:-}" != 1 ]; then
+    skip "Non-deterministic test will fail if we are unlucky"
+fi
+
 COMMIT_SIGN=""
 if has_gpgme; then
     COMMIT_SIGN="--gpg-homedir=${TEST_GPG_KEYHOME} --gpg-sign=${TEST_GPG_KEYID_1}"
